pkg/reconciler/instances/istio/action.go

```diff
@@ -225,8 +235,11 @@ func canUpdate(ver actions.IstioVersion, logger *zap.SugaredLogger) bool {
 	dataPlaneVsTarget := targetHelperVersion.compare(&dataPlaneHelperVersion)
 
 	if pilotVsTarget == -1 || dataPlaneVsTarget == -1 {
-		logger.Errorf("Downgrade detected from pilot: %s and data plane: %s to version: %s - finishing...", ver.PilotVersion, ver.DataPlaneVersion, ver.TargetVersion)
-		return false
+		if !targetHelperVersion.isDowngradePermitted(&pilotHelperVersion) || !targetHelperVersion.isDowngradePermitted(&dataPlaneHelperVersion) {
+			logger.Errorf("Downgrade detected from pilot: %s and data plane: %s to version: %s - finishing...", ver.PilotVersion, ver.DataPlaneVersion, ver.TargetVersion)
+			return false
+		}
+		logger.Infof("Valid Downgrade detected from pilot: %s and data plane: %s to version: %s", ver.PilotVersion, ver.DataPlaneVersion, ver.TargetVersion)
```

Comment:
I would return "true" here immediately, in that way you don't have to reason what the logic in lines 245-249 does.